src/m365/spo/commands/web/web-roleinheritance-break.spec.ts

```diff
@@ -0,0 +1,246 @@
+    assert.notStrictEqual(actual, true);
+  });
+
+  // it('passes validation if the url option is a valid SharePoint site URL', async () => {
+  //   const actual = await command.validate({ options: { webUrl: 'https://contoso.sharepoint.com' } }, commandInfo);
+  //   assert.strictEqual(actual, true);
+  // });
+
+  it('passes validation if the url option is a valid SharePoint site URL', async () => {
+    const actual = await command.validate({
+      options: {
+        webUrl: "https://contoso.sharepoint.com"
+      }
+    }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('should prompt before deleting when confirmation argument not passed', async () => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      requests.push(opts);
+      if ((opts.url as string).indexOf('/_api/web/breakroleinheritance(true)') > -1) {
+        return Promise.resolve(true);
+      }
+      return Promise.reject('Invalid request');
+    });
+    await command.action(logger, { options: { webUrl: "https://contoso.sharepoint.com" } });
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+    assert(promptIssued);
+  });
+
+  it('deletes web successfully when prompt confirmed', async () => {
+    // Delete web
+    sinon.stub(request, 'post').callsFake((opts) => {
+      requests.push(opts);
+      if ((opts.url as string).indexOf('_api/web') > -1) {
+        return Promise.resolve(true);
+      }
+      return Promise.reject('Invalid request');
+    });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+
+    await command.action(logger, {
+      options: {
+        webUrl: "https://contoso.sharepoint.com/subsite"
+      }
+    });
+    let correctRequestIssued = false;
+    requests.forEach(r => {
+      if (r.url.indexOf(`/_api/web/breakroleinheritance(true)`) > -1 &&
+        r.headers['accept'] === 'application/json;odata=nometadata') {
+        correctRequestIssued = true;
+      }
+    });
+    assert(correctRequestIssued);
+  });
+
+  it('Breack role inheritance successfully when prompt confirmed', async () => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      requests.push(opts);
+      if ((opts.url as string).indexOf('/_api/web/breakroleinheritance(true)') > -1) {
+        return Promise.resolve(true);
+      }
+      return Promise.reject('Invalid request');
+    });
+
+    await command.action(logger, {
+      options: {
+        debug: true,
+        webUrl: 'https://contoso.sharepoint.com',
+        confirm: true
+      }
+    });
+    let correctRequestIssued = false;
+    requests.forEach(r => {
+      if (r.url.indexOf(`/_api/web/breakroleinheritance(true)`) > -1 &&
+        r.headers['accept'] === 'application/json;odata=nometadata') {
+        correctRequestIssued = true;
+      }
+    });
+    assert(correctRequestIssued);
+  });
+
```

Comment:
Let's add another test to ensure the inheritance is not broken when the users declines.